dipy/reconst/dki_micro.py

```diff
@@ -0,0 +1,507 @@
+            params = params_all_mask.reshape(out_shape)
+        else:
+            params = np.zeros(data.shape[:-1] + (params_all_mask.shape[-1],))
+            params[mask, :] = params_all_mask
+
+        return KurtosisMicrostructuralFit(self, params)
+
+    def predict(self, params, S0=1.):
+        """ Predict a signal for the DKI microstructural model class instance
+        given parameters.
+
+        Parameters
+        ----------
+        params : ndarray (x, y, z, 40) or (n, 40)
+            All parameters estimated from the diffusion kurtosis
+            microstructural model.
+            Parameters are ordered as follows:
+                1) Three diffusion tensor's eigenvalues
+                2) Three lines of the eigenvector matrix each containing the
+                   first, second and third coordinates of the eigenvector
+                3) Fifteen elements of the kurtosis tensor
+                4) Six elements of the hindered diffusion tensor
+                5) Six elements of the restricted diffusion tensor
+                6) Axonal water fraction
+        S0 : float or ndarray (optional)
+            The non diffusion-weighted signal in every voxel, or across all
+            voxels. Default: 1
+        """
+        return dkimicro_prediction(params, self.gtab, S0)
+
+
+class KurtosisMicrostructuralFit(DiffusionKurtosisFit):
+    """ Class for fitting the Diffusion Kurtosis Microstructural Model """
+
+    def __init__(self, model, model_params):
+        """ Initialize a KurtosisMicrostructural Fit class instance.
+
+        Parameters
+        ----------
+        model : DiffusionKurtosisModel Class instance
+            Class instance containing the Diffusion Kurtosis Model for the fit
+        model_params : ndarray (x, y, z, 40) or (n, 40)
+            All parameters estimated from the diffusion kurtosis
+            microstructural model.
+            Parameters are ordered as follows:
+                1) Three diffusion tensor's eigenvalues
+                2) Three lines of the eigenvector matrix each containing the
+                   first, second and third coordinates of the eigenvector
+                3) Fifteen elements of the kurtosis tensor
+                4) Six elements of the hindered diffusion tensor
+                5) Six elements of the restricted diffusion tensor
+                6) Axonal water fraction
+
+        Notes
+        -----
+        Since this model is an extension of DKI, class instance is defined
+        as subclass of the DiffusionKurtosisFit from dki.py
+        """
+        DiffusionKurtosisFit.__init__(self, model, model_params)
+
+    @property
+    def awf(self):
+        """ Returns the volume fraction of the restricted diffusion compartment
+        also known as axonal water fraction.
+        """
+        return self.model_params[..., 27]
+
+    @property
+    def restricted_evals(self):
+        """ Returns the eigenvalues of the restricted diffusion compartment.
```

Comment:
Put a note here mentioning that this is called "intracellular" (or was it "extracellular"?) in the Fieremans paper.